File: example/ios/Runner/Biz/PowerImageFileImageLoader.m
//
//  MyFileImageLoader.m
//  Runner
//
//  Created by 王振辉 on 2021/8/6.
//

#import "PowerImageFileImageLoader.h"
#import <UIKit/UIKit.h>

@implementation PowerImageFileImageLoader
- (void)handleRequest:(PowerImageRequestConfig *)requestConfig completed:(PowerImageLoaderCompletionBlock)completedBlock {
    
    UIImage *image = [[UIImage alloc] initWithContentsOfFile:requestConfig.srcString];

    if (image) {
        completedBlock([PowerImageResult successWithImage:image]);
    } else {
        completedBlock([PowerImageResult failWithMessage:@"UIImage initWithContentsOfFile nil"]);
    }
}
@end

File: example_kotlin_swift/ios/Runner/Loaders/PowerImageFileImageLoader.swift
//
//  PowerImageFileImageLoader.swift
//  Runner
//
//  Created by wjr on 2022/5/30.
//
import power_image

class PowerImageFileImageLoader: NSObject, PowerImageLoaderProtocol{
    
    func handleRequest(_ requestConfig: PowerImageRequestConfig!, completed completedBlock: PowerImageLoaderCompletionBlock!) {
        
        let image = UIImage(contentsOfFile: requestConfig.srcString())
        
        if let image = image {
            completedBlock(PowerImageResult.success(with: image))
        }else{
            completedBlock(PowerImageResult.fail(withMessage: "PowerImageFileImageLoaderError!"))
        }
    }
}
